pyexchange/fix.py

```diff
@@ -258,16 +258,19 @@ def _run_session(self):
     async def _session_proc(self):
         (address, port) = tuple(self.endpoint.split(':'))
 
-        if self.certs is not None:
-            self.ssl_context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
-            self.ssl_context.load_cert_chain(certfile=self.certs['client_cert'], keyfile=self.certs['client_key'])
-            self.ssl_context.check_hostname = False
-            self.ssl_context.verify_mode = ssl.CERT_NONE
-            self.reader, self.writer = await asyncio.open_connection(address, port, loop=self.session_loop, ssl=self.ssl_context)
-        else:
-            self.reader, self.writer = await asyncio.open_connection(address, port, loop=self.session_loop)
-
-        self.connection_state = FixConnectionState.CONNECTED
+        try:
+            if self.certs is not None:
+                self.ssl_context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
+                self.ssl_context.load_cert_chain(certfile=self.certs['client_cert'], keyfile=self.certs['client_key'])
+                self.ssl_context.check_hostname = False
+                self.ssl_context.verify_mode = ssl.CERT_NONE
+                self.reader, self.writer = await asyncio.open_connection(address, port, loop=self.session_loop, ssl=self.ssl_context)
+            else:
+                self.reader, self.writer = await asyncio.open_connection(address, port, loop=self.session_loop)
+
+            self.connection_state = FixConnectionState.CONNECTED
+        except Exception as e:
```

Comment:
Not happy with using base exception; but i'm not yet sure what type of exception could be thrown

Reply:
Fair enough.  And sometimes it's acceptable to put a catch-all at the root level of a thread.